ksql-parser/src/main/antlr4/io/confluent/ksql/parser/SqlBase.g4

```diff
@@ -46,6 +46,7 @@ statement
     | PRINT (identifier| STRING) printClause                                #printTopic
     | (LIST | SHOW) QUERIES EXTENDED?                                       #listQueries
     | TERMINATE QUERY? identifier                                           #terminateQuery
+    | TERMINATE ALL                                                         #terminateQuery
```

Comment:
what about changing the semantics of `TERMINATE <foo>` to use regex instead? That seems much more powerful and we don't need to introduce any new reserved words.

Reply:
I kind of feel like that could be a slippery slope. e.g. do we introduce regex semantics elsewhere in the grammar? How do users know where they can use regexes and where they can't?

In terms of selectively performing an operation on a number of objects based on a condition, other databases often use catalogs. e.g.,

```sql
SELECT terminate(query_id) FROM query_catalog WHERE <condition>
```
Which I think is a great pattern but obviously not possible for us.

Personally my vote here would be to completely do away with `TERMINATE` and just make `DROP` drop the object and its underlying query. It doesn't seem like `TERMINATE` is useful for users, because as far as I know a terminated query can't be restarted anyways. So it seems to serve as a workaround for the problem of not being able to `DROP CASCADE` a stream or table.

Reply:
The slippery slope is fair... that's why I'd like to see these things discussed before adding it 😂 

> In terms of selectively performing an operation on a number of objects based on a condition, other databases often use catalogs. e.g.,
> 
> ```sql
> SELECT terminate(query_id) FROM query_catalog WHERE <condition>
> ```
> 
> Which I think is a great pattern but obviously not possible for us.

(off topic) The wheels in my mind are turning 😈 I think we should expose a query_catalog as a KSQL table! It's basically a materialization on the command topic. I'm about to create a ticket for this because I think we need to change the command topic to have table semantics anyway.

Reply:
Proper catalogs would be amazing and would open up all kinds of other operational possibilities. Catalogs also really empower tools, automation etc.

Reply:
#3945

Reply:
unresolving so that this thread is more visible when people refer back to it.